railties/lib/rails/command.rb

```diff
@@ -33,13 +34,15 @@ def invoke(full_namespace, args = [], **config)
 
         if char = namespace =~ /:(\w+)$/
           command_name, namespace = $1, namespace.slice(0, char)
+        elsif namespace.blank?
+          command_name, namespace = "help", "help"
+        elsif HELP_MAPPINGS.include?(namespace)
+          command_name, namespace = "help_extended", "help"
+        elsif VERSION_MAPPINGS.include?(namespace)
+          command_name, namespace = "version", "version"
         else
           command_name = namespace
         end
```

Comment:
If `HELP_MAPPINGS` and `VERSION_MAPPINGS` were `Set`s or `Regexp`s, we could write:

```suggestion
        case namespace
        when /^(.+):(\w+)$/
          namespace, command_name = $1, $2
        when ""
          namespace, command_name = "help", "help"
        when HELP_MAPPINGS
          namespace, command_name = "help", "help_extended"
        when VERSION_MAPPINGS
          namespace, command_name = "version", "version"
```

Also, it might be good to factor this out into a method like `split_command_name` or `split_namespace_and_command_name` which would return a `[namespace, command_name]` pair.

Reply:
Ah, that looks very nice. 👍 
I've extracted it to a `split_namespace` method.